### PR TITLE
Fixes IE browser back issue when clearing selections using backspace key

### PR DESCRIPTION
**Pull request by ToreOlavKristiansen**
This pull request includes a

- [X] Bug fix
- [ ] New feature
- [ ] Translation

The following changes were made

I changed allowClear.js to:
- Fix IE browser back issue when clearing selections using backspace key.
- Fix issue with not being able to start typing for both IE and Chrome after clearing using backspace or delete key.

If this is related to an existing ticket, include a link to it as well.
#3168

### CODE REVIEW

**Comment by alexweissman** [comment]
@cordmata @dmarginian can you please take a look at this PR, and see if it solves your other issues with IE?

File: dist/js/select2.full.js [hunk 1]
```diff
@@ -2019,7 +2027,13 @@ S2.define('select2/selection/search',[
 
     this.resizeSearch();
     if (searchHadFocus) {
-      this.$search.focus();
+      var isTagInput = this.$element.find('[data-select2-tag]').length;
+      if (isTagInput) {
```
Comment (TRoorda) [inline dist/js/select2.full.js]: To make this work in IE11 with multiple tag support, I had to combine this code with the code presented in a comment on https://github.com/select2/select2/issues/4398:
```var isTagInput = this.$element.find('[data-select2-tag]').length;
if (isTagInput) {
    // fix IE11 bug where tag input lost focus
    window.setTimeout(function() {
        self.$element.focus();
    }, 0);
} else {
    window.setTimeout(function() {
        self.$search.focus();
    }, 0);
}

**Comment by caohanh1502** [comment]
I have a problem here. When I apply this fix, the browser would focus on the select2 again if I click on an input while the dropdown selection is still opened. It does not happen in v4.0.3.
I also tested on the newest dev version 4.0.6-rc.1 with Chrome 68 and IE 11, and the same things happen.

Codepen: [https://codepen.io/caohanh1502/pen/MBRQWR](https://codepen.io/caohanh1502/pen/MBRQWR)
Debug view for IE 11: [https://s.codepen.io/caohanh1502/debug/MBRQWR/vPrKKzzxBPKr](https://s.codepen.io/caohanh1502/debug/MBRQWR/vPrKKzzxBPKr)

Click on the select2 to open the dropdown selection, then click on the input of datetimepicker. Result:
- Chrome: The datetimepicker is shown but it hides immediately. The focus is set to the select2, and I cannot type anything into the input.
- IE 11: The datetimepicker does not shown (I did not see the div appear in the IE's DOM explorer). The focus is set to the select2, but I can type into the input.
- (Same with the normal input).

**Comment by pedrofurtado** [comment]
https://github.com/select2/select2/pull/3394

File: dist/js/select2.full.js (hunk 1 above)
Comment (andreikondratev) [reply dist/js/select2.full.js]: This one worked for me

**Comment by kevin-brown** [comment]
Unfortunately this pull request contains files in the `dist` folder, and as a result there is a merge conflict. Because this pull request is also pretty old (our fault), we are closing this off.

If you are still interested in making this contribution, feel free to remove the `dist` changes from this pull request and leave a comment asking for this to be re-reviewed. Or make a new pull request from the same branch, and we'll try to get around to reviewing it this time.

**Pull request closed without merging**